backends/qualcomm/qnn_quantizer.py

```diff
@@ -733,9 +837,199 @@ def _annotate_upsample2d(self, gm: torch.fx.GraphModule) -> None:
                 _annotated=True,
             )
 
+    def _annotate_embedding(self, gm: torch.fx.GraphModule) -> None:
+        op_sources = [torch.nn.modules.sparse.Embedding, torch.nn.modules.Embedding]
+        qconfig = self._get_quant_config(op_sources)
+        embedding_partitions = get_source_partitions(gm.graph, op_sources)
+
+        embedding_partitions = list(itertools.chain(*embedding_partitions.values()))
+        for embedding_partition in embedding_partitions:
+            embedding_node = embedding_partition.output_nodes[0]
+
+            weight = embedding_node.args[0]
+
+            input_qspec_map = {}
+            input_qspec_map[weight] = qconfig.input_activation
+
+            embedding_node.meta[QUANT_ANNOTATION_KEY] = QuantizationAnnotation(
+                input_qspec_map=input_qspec_map,
+                output_qspec=SharedQuantizationSpec((weight, embedding_node)),
+                _annotated=True,
```

Comment:
Do you have embedding support in qnn? If so does it produce int8 quantized embedding?

Reply:
We will use gather OP to replace it. It will produce quantized result if weight is quantized.

Reply:
But I thought somewhere you said embedding runs on CPU

Reply:
Yes, due to a regression for embedding op, we fall back to executorch via partitioner now. I think it will run on CPU.

Reply:
So if you are quantizing this just get weights quantized, but running it on CPU, then it will dequantize weights, run embedding op and quantize output again. Just want to highlight this. Is this what you want?

Reply:
Yes, this is currently the case. Once our regression is fixed, we can run on HTP.